src/main/java/org/mockito/plugins/InstantiatorProvider.java

```diff
@@ -4,7 +4,7 @@
  */
 package org.mockito.plugins;
 
-import org.mockito.internal.creation.instance.Instantiator;
+import org.mockito.creation.instance.Instantiator;
```

Comment:
This is a backwards incompatible change.

Reply:
But we regard `internal` as an exception to our backwards compatibility requirement right?

Reply:
@mockitoguy I see it like this:

Having a class marked as internal means that using this class might break if mockito is updated. Hence anybody using Instantiator should be aware that this might break. Hence I think breaking people is just the expected behavior.

Of course InstantiatorProvider is a public API and upgrading mockito should not break this API. I think there is only one use case of InstantiatorProvider without using internal APIs. It is calling getInstantiator and compare it to null or an existing object. This still works, even after my move.

In summary: Yes I am braking everybody using of the internal class. I don't find a use case where I break anyone using the public API.

Reply:
>But we regard internal as an exception to our backwards compatibility requirement right?

@TimvdLippe, yes. We have few exceptions from our standard rule, where internal API leaked and is considered public. The other exception is [VerificationData](https://github.com/mockito/mockito/blob/release/2.x/src/main/java/org/mockito/internal/verification/api/VerificationData.java) type that lives in "internal" package but is considered public (see the Javadoc).

Reply:
>In summary: Yes I am braking everybody using of the internal class. I don't find a use case where I break anyone using the public API.

@moltmann, you are right that it is not a breaking change provided that InstantiatorProvider is only used to null-check result from ```getInstantiator()``` method. Problem is that our "plugins" API contract tells users to provide implementations of plugin interfaces, such as ```InstantiatorProvider```. So anyone who uses ```InstantiatorProvider``` would actually provide his own implementation, and his code would break if we ship this PR as-is.

If you are concerned about using "internal" class that we consider public, what do you think about option b) (create new type, make the change compatible)?

Reply:
If there is precedent for a internal-quite-not-internal API, I can mark it with the same statement as VerificationData.

The statement in the JavaDoc of VerificationData states that the class will be moved though. Will this actually happen?